app/src/main/java/org/schabi/newpipe/util/URLHandler.java

```diff
@@ -0,0 +1,113 @@
+package org.schabi.newpipe.util;
+
+import android.content.Context;
+
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.linkhandler.LinkHandlerFactory;
+import org.schabi.newpipe.extractor.stream.StreamInfo;
+import org.schabi.newpipe.player.playqueue.PlayQueue;
+import org.schabi.newpipe.player.playqueue.SinglePlayQueue;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+public final class URLHandler {
+
+    private URLHandler() {
+    }
+
+    /**
+     * Check if an URL can be handled in NewPipe.
+     * <p>
+     * This method will check if the provided url can be handled in NewPipe or not. If this is a
+     * service URL with a timestamp, the popup player will be opened.
+     * <p>
+     * The timestamp param accepts two integers, corresponding to two timestamps types:
+     * 0 for {@code &t=} (used for timestamps in descriptions),
+     * 1 for {@code #timestamp=} (used for timestamps in comments).
+     * Any other value of this integer will return false.
+     *
+     * @param context       the context to be used
+     * @param url           the URL to check if it can be handled
+     * @param timestampType the type of timestamp
+     * @return true if the URL can be handled by NewPipe, false if it cannot
+     */
+    public static boolean handleUrl(final Context context, final String url, final int timestampType) {
```

Comment:
While you are at it: please rename this method to `canHandleUrl` to make it more clear what this method is supposed to do.

Reply:
Yeah, this makes sense. I will change it later, when I have some time.